src/drivers/redis.ts

```diff
@@ -56,25 +63,67 @@ export default defineDriver((opts: RedisOptions) => {
   const p = (...keys: string[]) => joinKeys(base, ...keys); // Prefix a key. Uses base for backwards compatibility
   const d = (key: string) => (base ? key.replace(base, "") : key); // Deprefix a key
 
+  const getItem = async (key: string) => {
+    const value = await getRedisClient().get(p(key));
+    return value ?? null;
+  };
+
+  const setItem = async (
+    key: string,
+    value: any,
+    tOptions: TransactionOptions
+  ) => {
+    const ttl = tOptions?.ttl ?? opts.ttl;
+    if (ttl) {
+      await getRedisClient().set(p(key), value, "EX", ttl);
+    } else {
+      await getRedisClient().set(p(key), value);
+    }
+  };
+
+  const getItemRaw = async (key: string) => {
+    const value = await getRedisClient().getBuffer(p(key));
+    return value ?? null;
+  };
+
+  const setItemRaw = async (
+    key: string,
+    value: Uint8Array,
+    tOptions: TransactionOptions
+  ) => {
+    let valueToSave: Buffer;
+    if (value instanceof Uint8Array) {
+      if (value instanceof Buffer) {
+        valueToSave = value;
+      } else {
+        valueToSave = Buffer.copyBytesFrom(
+          value,
+          value.byteOffset,
+          value.byteLength
+        );
+      }
+    } else {
+      throw createError(DRIVER_NAME, "Expected Buffer or Uint8Array");
+    }
+
+    await setItem(key, valueToSave, tOptions);
```

Comment:
For fallback, while core ensures `values` is string, here we cannot assume the same